cmd/oras/root/cp.go

```diff
@@ -257,9 +257,15 @@ func prepareCopyOption(ctx context.Context, src oras.ReadOnlyGraphTarget, dst or
 		return opts, nil
 	}
 
-	referrers, err := graph.FindPredecessors(ctx, src, index.Manifests, opts)
-	if err != nil {
-		return opts, err
+	// do breadth first search to find all levels of child referrers
+	var referrers []ocispec.Descriptor
+	for descs := index.Manifests; len(descs) > 0; {
```

Comment:
nit: For better readability, how about something like this:
```go
	nodesToFind := index.Manifests
	for len(nodesToFind) > 0 {
	      // do something...
	}
```

Reply:
TBH, the one-line style confused me at the first glance...

Reply:
We can go with what @Wwwsylvia has suggested :)